src/content/app/species/state/sidebar/speciesSidebarSlice.ts

```diff
@@ -130,10 +145,74 @@ const speciesPageSidebarSlice = createSlice({
       updateStateForGenome(state, action.payload.genomeId, {
         payload: action.payload.sidebarPayload
       });
+    },
+
+    updateSpeciesSidebarModalForGenome(
+      state,
+      action: PayloadAction<{
+        activeGenomeId: string;
+        data: Partial<StateForGenome>;
+      }>
+    ) {
+      const { activeGenomeId, data } = action.payload;
+
+      state[activeGenomeId] = {
+        ...state[activeGenomeId],
+        ...data
+      };
     }
   }
 });
 
-export const { toggleSidebar } = speciesPageSidebarSlice.actions;
+export const openSpeciesSidebarModal =
```

Comment:
The only purpose of this thunk is to get the active genome id out of the state and pass it to another redux action. If you change the payload of `openSpeciesSidebarModal ` and `closeSpeciesSidebarModal` such that it includes the genome id; then you won't need these two thunks, and could just have two reducers on the slice.

Reply:
Can someone help me with this?